devices/IFrameTransformToIWear/src/IFrameTransformToIWear.cpp

```diff
@@ -0,0 +1,518 @@
+    wearable::sensor::SensorType wearableSensorType;
+    std::string rootFrameID;
+
+    std::vector<std::string> frameIDs;
+};
+
+// Class that provides each frame information, and utilites
+// to map its data to containers compatible with IWear
+class IFrameTransformHandler
+{
+public:
+    IFrameTransformHandler(const std::string targetFrame,
+                           const std::string rootFrame,
+                           yarp::dev::IFrameTransform* frameTransformInterface)
+        : m_targetFrame(targetFrame)
+        , m_rootFrame(rootFrame)
+        , m_frameTransformInterface(frameTransformInterface)
+        , m_matrixBuffer(yarp::sig::Matrix(4, 4)){};
+
+    wearable::sensor::SensorStatus getStatus()
+    {
+        if (m_frameTransformInterface->canTransform(m_targetFrame, m_rootFrame)) {
+            return SensorStatus::Ok;
+        }
+        else {
+            return SensorStatus::Error;
+        }
+    }
+
+    bool getTransform(wearable::Quaternion& quaternion, wearable::Vector3& position)
+    {
+
+        if (!m_frameTransformInterface->getTransform(m_targetFrame, m_rootFrame, m_matrixBuffer)) {
+            return false;
+        }
+
+        position[0] = m_matrixBuffer(0, 3);
+        position[1] = m_matrixBuffer(1, 3);
+        position[2] = m_matrixBuffer(2, 3);
+
+        for (size_t r = 0; r < 3; r++) {
+            for (size_t c = 0; c < 3; c++) {
+                m_rotationMatrixBuffer[r][c] = m_matrixBuffer(r, c);
+            }
+        }
+
+        quaternion = wearable::utils::rotationMatrixToQuaternion(m_rotationMatrixBuffer);
+        return true;
+    }
+
+private:
+    yarp::dev::IFrameTransform* m_frameTransformInterface = nullptr;
+    std::string m_rootFrame, m_targetFrame;
+    yarp::sig::Matrix m_matrixBuffer;
+    wearable::Matrix3 m_rotationMatrixBuffer;
+};
+
+class IFrameTransformToIWear::Impl
+{
+public:
+    bool firstRun = true;
+    mutable std::recursive_mutex mutex;
+
+    TimeStamp timestamp;
+    ParsedOptions options;
+
+    yarp::dev::IFrameTransform* frameTransformInterface = nullptr;
+
+    template <typename T>
+    using SensorsMap = std::map<std::string, SensorPtr<T>>;
+
+    // Sensors stored for exposing wearable::IWear
+    class PoseSensor;
+    SensorsMap<PoseSensor> poseSensorsMap;
+
+    template <typename SensorType>
+    bool isSensorAvailable(const std::string& name, const SensorsMap<SensorType>& map)
+    {
+        if (map.find(name) == map.end()) {
+            return false;
+        }
+        return true;
+    }
+};
+
+// ================================
+// WEARABLE SENSORS IMPLEMENTATIONS
+// ================================
+
+class IFrameTransformToIWear::Impl::PoseSensor : public wearable::sensor::IPoseSensor
+{
+public:
+    // ------------------------
+    // Constructor / Destructor
+    // ------------------------
+    PoseSensor(
+        IFrameTransformHandler* frameTransformHandler,
+        const wearable::sensor::SensorName aName,
```

Comment:
```suggestion
        const wearable::sensor::SensorName& aName,
```

Reply:
what's the benefit of defining `aName` as a reference ?